test/Microsoft.AspNet.Razor.Test/SourceLocationTest.cs

```diff
@@ -14,9 +15,238 @@ public void ConstructorWithLineAndCharacterIndexSetsAssociatedProperties()
             var loc = new SourceLocation(0, 42, 24);
 
             // Assert
+            Assert.Null(loc.FilePath);
             Assert.Equal(0, loc.AbsoluteIndex);
             Assert.Equal(42, loc.LineIndex);
             Assert.Equal(24, loc.CharacterIndex);
         }
+
+        [Fact]
+        public void Constructor_SetsFilePathAndIndexes()
+        {
+            // Arrange
+            var filePath = "some-file-path";
+            var absoluteIndex = 133;
+            var lineIndex = 23;
+            var characterIndex = 12;
+
+            // Act
+            var sourceLocation = new SourceLocation(filePath, absoluteIndex, lineIndex, characterIndex);
+
+            // Assert
+            Assert.Equal(filePath, sourceLocation.FilePath);
+            Assert.Equal(absoluteIndex, sourceLocation.AbsoluteIndex);
+            Assert.Equal(lineIndex, sourceLocation.LineIndex);
+            Assert.Equal(characterIndex, sourceLocation.CharacterIndex);
+        }
+
+        [Fact]
+        public void GetHashCode_ReturnsHashCode_UsingAbsoluteIndex()
+        {
+            // Arrange
+            var sourceLocationA = new SourceLocation(10, 3, 4);
+            var sourceLocationB = new SourceLocation(10, 45, 8754);
+
+            // Act
+            var hashCodeA = sourceLocationA.GetHashCode();
+            var hashCodeB = sourceLocationB.GetHashCode();
+
+            // Assert
+            Assert.Equal(hashCodeA, hashCodeB);
+        }
+
+        [Fact]
+        public void GetHashCode_ReturnsHashCode_UsingFilePathAndAbsoluteIndex_WhenFilePathIsNonNull()
+        {
+            // Arrange
+            var sourceLocationA = new SourceLocation("some-path", 3, 53, 94);
+            var sourceLocationB = new SourceLocation("some-path", 3, 37, 46);
```

Comment:
@NTaylorMullen's comment hasn't been addressed.  should be something like
``` c#
var sourceLocationA = new SourceLocation(3, 53, 94);
var sourceLocationB = new SourceLocation("some-path", 3, 53, 94);
```
then verify the hash codes are _different_